js/run.js

```diff
@@ -179,13 +179,17 @@ function parseOutput(res, o, oTitle)
 }
 
 // wraps a unittest into a runnable script
-function wrapIntoMain(code) {
+function wrapIntoMain(code, compile) {
     var currentPackage = $('body')[0].id;
 
     // dynamically wrap into main if needed
     if (code.indexOf("void main") >= 0 || code.indexOf("int main") >= 0) {
         return code;
     }
+    else if (compile) {
+        // avoid missing main link error
+        return code + "\nvoid main() {}";
```

Comment:
The Compile button does run the code too (perhaps I should add a tooltip to state that). That's useful for examples the user may want to execute but that the test suite doesn't need to (e.g. writeln).

Compile currently doesn't auto import stdio, though perhaps it should.

Reply:
I think compile shouldn't auto-import std.stdio as usually that's not needed.